### PR TITLE
feat: defsec k8s scanner integration

### LINKED ISSUE #57: Incorporate Kubernetes policy scanner into trivy-operator
The motivation for this task is to eat aqua open source dog food, meaning have a unified kubernetes policy scanner lib used in the builtin config audit scanner (same as used in other projects).

In this task, trivy-operator will incorporate the new Kubernetes policy scanner introduced by Defsec and it will replace the embedded policy lib used to evaluate configAudit rego scripts.

here is the reference to the new scanner : https://github.com/aquasecurity/defsec/tree/master/pkg/scanners/kubernetes

### PR DESCRIPTION
**Pull request by chen-keinan**
Signed-off-by: chenk <hen.keinan@gmail.com>

## Description
replace trivy-operator implementation for validating policies with DefSec scanner (same one used in Trivy)

## Related issues
- Close #57 

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: pkg/policy/policy.go [hunk 1]
```diff
@@ -187,6 +109,17 @@ func (p *Policies) ModulesByKind(kind string) (map[string]string, error) {
 	}
 	return modules, nil
 }
+func (p *Policies) ModulePolicyByKind(kind string) ([]string, error) {
+	modByKind, err := p.ModulesByKind(kind)
+	if err != nil {
+		return nil, err
+	}
+	policy := make([]string, 0)
```
Comment (josedonizetti) [inline pkg/policy/policy.go:117]:
```suggestion
	policy := make([]string, 0, len(modByKind))
```

File: pkg/policy/policy.go [hunk 2]
```diff
@@ -204,182 +137,57 @@ func (p *Policies) Applicable(resource client.Object) (bool, string, error) {
...
+		return nil, err
 	}
-	return values, len(rs) == len(values)
-}
-
-func requiredStringValue(values map[string]interface{}, key string) (string, error) {
-	value, ok := values[key]
-	if !ok {
-		return "", fmt.Errorf("required key not found: %s", key)
+	inputResource, err := json.Marshal(resource)
```
Comment (josedonizetti) [inline pkg/policy/policy.go:162]: Can we run json marshal before declaring the const on line 147? if we can't marshal for some reason, there is no need get the policies and to create a memoryfs.

File: pkg/policy/policy.go [hunk 3]
```diff
@@ -204,182 +137,57 @@ func (p *Policies) Applicable(resource client.Object) (bool, string, error) {
...
-	for _, value := range values {
-		message, err := NewMessage(value)
-		if err != nil {
-			return nil, err
+func createPolicyInputFS(memfs *memoryfs.FS, folderName string, fileData []string, ext string) error {
+	if err := memfs.MkdirAll(filepath.Base(folderName), 0o700); err != nil {
+		return err
+	}
+	for index, file := range fileData {
+		if err := memfs.WriteFile(path.Join(folderName, fmt.Sprintf("%s_%d.%s", "file", index, ext)), []byte(file), 0o644); err != nil {
```
Comment (josedonizetti) [inline pkg/policy/policy.go:188]:
```suggestion
		if err := memfs.WriteFile(path.Join(folderName, fmt.Sprintf("file_%d.%s", index, ext)), []byte(file), 0o644); err != nil {
```

**Review by josedonizetti — APPROVED** [review]
@chen-keinan couple suggestions, nothing major. Feel free to change it or not.

File: pkg/policy/policy.go (hunk 2 above)
Comment (chen-keinan) [reply pkg/policy/policy.go:162]: I prefer to declare `inputResource` close to its usage L167

**Comment by josedonizetti** [comment]
@chen-keinan Also, not sure if you mind this PR getting merged for release v0.0.6 or you rather wait for next release. Up to you, just mentioning because we are releasing  v0.0.6 tmr, and depending on what you decide you need to wait to merge.

**Pull request merged**